apps/Beacon.py

```diff
@@ -0,0 +1,130 @@
+        b'\x92\xc1?\r\xc1\x92\xc1?\r\xc1\x92\xc1?\r\xc1\x92'
+        b'\xc1?\r\xc1\x92\xc1?\r\xc1\x92\xc1?\x0c\xc1\x94\xc1'
+        b'?\x0b\xd6?\x0b\xc1@zT\xc1?\x0b\xc1T\xc1?'
+        b'\x0b\xc1T\xc1?\x0b\xc1T\xc1?\n\xc1V\xc1?\t'
+        b'\xc1V\xc1?\t\xc1V\xc1?\t\xc1V\xc1?\t\xc1'
+        b'V\xc1?\t\xc1V\xc1?\t\xc1V\xc1?\x08\xda?'
+        b'\x07\xc1\x98\xc1?\x07\xc1\x98\xc1?\x07\xc1\x98\xc1?\x07'
+        b'\xc1\x98\xc1?\x07\xc1\x98\xc1?\x06\xc1\x9a\xc1?\x05\xc1'
+        b'\x9a\xc1?\x05\xc1\x9a\xc1?\x05\xdc?\xff\x04'
+    )
+
+
+    def __init__(self):
+        self._checkbox = wasp.widgets.Checkbox(10, 45, "Enable beacon")
+        self._slider_current = wasp.widgets.Slider(4, 10, 110, 0x27e4)
+        self._slider_wait_time = wasp.widgets.Slider(8, 10, 180)
+        self._reg_enable = 0x00 # off, 800ms, part of 12.5mA
+        self._reg_pdriver = 0x2e #0x4e # part of 12.5mA, PON=1, magic
+
+    def foreground(self):
+        wasp.system.bar.clock = True
+        self._draw()
+        wasp.system.request_event(wasp.EventMask.TOUCH)
+
+    def _draw(self):
+        draw = wasp.watch.drawable
+        draw.fill()
+        wasp.system.bar.draw()
+        self._checkbox.draw()
+        draw.string("Intensity:", 10, 85)
+        self._slider_current.draw()
+        draw.string("Frequency:", 10, 155)
+        self._slider_wait_time.draw()
+        self._update()
+
+    def touch(self, event):
+        updated = self._checkbox.touch(event)
+        if event[2] >= 180:
+            updated |= self._slider_wait_time.touch(event)
+        elif event[2] >= 110:
+            updated |= self._slider_current.touch(event)
+        self._update()
+        if updated:
+            if self._checkbox.state:
+                self._enable_led()
+            else:
+                self._disable_led()
+
+    def _update(self):
+        wasp.system.bar.update()
+        self._checkbox.update()
+        self._slider_current.update()
+        self._slider_wait_time.update()
+
+        # Draw a dashed line representing intensity and frequency
+        # with thickness and separation of dashes
+        draw = wasp.watch.drawable
+        draw.fill(None, 10, 220, 227, 20)
+        x = 10
+        while x < 220:
+            wasp.watch.drawable.fill(0x27e4, x, 227, 8, (self._slider_current.value + 1) * 3)
+            x += (8 - self._slider_wait_time.value) * 8
+
+    def _enable_led(self):
+        self._reg_enable |= _ENABLE_HEN
+        if self._slider_current.value % 2:
+            self._reg_pdriver |= _PDRIVER_PDRIVE0
+        else:
+            self._reg_pdriver &= ~_PDRIVER_PDRIVE0
+        if self._slider_current.value >> 1:
+            self._reg_enable |= _ENABLE_PDRIVE1
+        else:
+            self._reg_enable &= ~_ENABLE_PDRIVE1
+        self._reg_enable = (self._reg_enable & (~_ENABLE_HWT)) | (self._slider_wait_time.value << 4)
+        wasp.watch.hrs.write_reg(_PDRIVER, self._reg_pdriver)
```

Comment:
I don't think direct calls from apps to read/write registers are the best way to do this. In general we want to reduce the number of places hrs3300 specific assumptions get burnt into apps.

Might be better to have an method to specify the LED intensity.

Reply:
Addressed